### DISCUSSION TITLE
Machine-wide installation on shared computers

### DISCUSSION THREAD

**Discussion by dolnack** [BHoM/BHoM#1431] [Q&A]
We're trying to install the BHoM in our computer labs, but it seems to default to per-user installs, and is not responding to the ALLUSERS=1 option when installing silently/via command line. Is there any way to install this for all users who will log into the machine?

**Comment by FraserGreenroyd** [comment]
Hi @dolnack there isn't currently a way to install the BHoM for all users, as the UI components act as plugins to their software which end up installing as local plugins only (e.g. the Grasshopper Plugin currently installs to `%AppData%/Grasshopper/Libraries/BH.UI.Grasshopper.ghlink`). However, the core functionality of the BHoM installs to `%ProgramData%/BHoM` which can be accessed by all users, though depending on computer set up it might become 'locked' to the user which installs it. It may be possible to copy the `ghlink` file and other associated UI files into individual user folders and have it work for all users as this generally doesn't require any admin rights.

Unfortunately, other than that, the BHoM installer has not been designed for multi-user installation at this time, but it is something we could add to the road map for future releases.
